File: response-service/src/main/java/fr/vandriessche/rallyeschema/responseservice/entities/ResponseFileSummary.java
package fr.vandriessche.rallyeschema.responseservice.entities;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class ResponseFileSummary {
	private String id;
	private Integer stage;
	private Integer page;
	private Integer team;
	private String processingStatus;
	private String processingError;
	private Boolean manualReviewRequired;
	private boolean availableForVerification;
	private boolean leasedByCurrentDevice;
}

File: rallye-schema-response-webapp/src/app/upload/models/response-file-summary.ts
export interface ResponseFileSummary {
  id: string;
  stage?: number;
  page?: number;
  team?: number;
  processingStatus?: string;
  processingError?: string;
  manualReviewRequired?: boolean;
  availableForVerification: boolean;
  leasedByCurrentDevice: boolean;
}

export interface ResponseFileSummaryPage {
  content: ResponseFileSummary[];
  totalElements: number;
  totalPages: number;
  number: number;
  size: number;
}
